src/session-adapters/internalAdapter.ts

```diff
@@ -64,65 +63,71 @@ export const createInternalAdapter: CreateInternalAdapter = ({
   res,
   adapter,
 }) => {
-  const sessionKey = sessionIdentifier(params.sessionKey)
-
   return {
     getSession: async ({ spaceId, userId }) => {
-      const session = await adapter.getItem({
-        req,
-        res,
-        clientId: params.clientId,
-        spaceId,
-        userId,
-        key: sessionKey,
-      })
-      if (!session) {
-        return undefined
-      }
       try {
-        return JSON.parse(session) as AppSession
-      } catch (err) {
+        const session = await adapter.getItem({
+          req,
+          res,
+          clientId: params.clientId,
+          spaceId,
+          userId,
+        })
+
+        if (!session) {
+          return undefined
+        }
+
+        return session
+      } catch (e) {
+        console.log('Retrieving the session failed: ', e)
         return undefined
       }
     },
 
     setSession: async ({ spaceId, userId, session }) => {
       try {
-        return await adapter.setItem({
+        await adapter.setItem({
           req,
           res,
           clientId: params.clientId,
           spaceId,
           userId,
-          key: sessionKey,
-          value: JSON.stringify(session),
+          value: session,
         })
-      } catch (err) {
+      } catch (e) {
+        console.log('Setting the session failed: ', e)
         return false
       }
     },
 
-    hasSession: ({ spaceId, userId }) =>
-      adapter.hasItem({
-        req,
-        res,
-        clientId: params.clientId,
-        spaceId,
-        userId,
-        key: sessionKey,
-      }),
+    hasSession: ({ spaceId, userId }) => {
+      try {
+        adapter.hasItem({
+          req,
+          res,
+          clientId: params.clientId,
+          spaceId,
+          userId,
+        })
+      } catch (e) {
+        console.log('Session could not be found: ', e)
+
+        return false
+      }
+    },
 
     removeSession: async ({ spaceId, userId }) => {
       try {
-        return await adapter.removeItem({
+        await adapter.removeItem({
```

Comment:
By the type definition above, removeSession should return Promise<boolean>, and we're missing it here.